pandas_gbq/schema/pandas_to_bigquery.py

```diff
@@ -164,7 +166,14 @@ def dtype_to_bigquery_field(name, dtype) -> Optional[schema.SchemaField]:
     return None
 
 
-def value_to_bigquery_field(name, value) -> Optional[schema.SchemaField]:
+def value_to_bigquery_field(
+    name, value, default_type=None
+) -> Optional[schema.SchemaField]:
+    # Set the SchemaField datatype to the given default_type if the value
```

Comment:
I know we are pushing to merge to enable further work, so the following is merely a #PREFERENCE or #SUGGESTION, not a blocker.

The code here works and tests pass so I don't feel like forcing us to fix something that isn't broken, but I wonder if we could clean up the code slightly (better readability, maintainability, efficiency).

* a type mapping removes some duplicate `isinstance()` calls and makes it easy to extend the code if we need to add checks against additional datatypes
* as seen in a comment that follows, we use `isinstance(v, str)` in two functions despite the fact that the outer function calls the inner function, which does a string check and then the outer function immediately does a second `isinstance()` check.

Will leave it up to you to decide whether any of these changes are reasonable in this PR.

```    if value is None:
        return schema.SchemaField(name, default_type)

    type_mapping = {
        str: "STRING",
        _BaseGeometry: "GEOGRAPHY",
    }

    for type_, field_type in type_mapping.items():
        if isinstance(value, type_):
            return schema.SchemaField(name, field_type)

    # For timezone-naive datetimes, the later pyarrow conversion to try and
    # learn the type add a timezone to such datetimes, causing them to be
    # recognized as TIMESTAMP type. We thus additionally check the actual data
    # to see if we need to overrule that and choose DATETIME instead.
    #
    # See: https://github.com/googleapis/python-bigquery/issues/985
    # and https://github.com/googleapis/python-bigquery/pull/1061
    # and https://github.com/googleapis/python-bigquery-pandas/issues/450
    if isinstance(value, datetime.datetime):
        return schema.SchemaField(
            name, "TIMESTAMP" if value.tzinfo else "DATETIME"
        )

    return None
```